.claude/hooks/pre-push-test.sh

```
@@ -0,0 +1,40 @@
+
+# git push 명령이 아니면 즉시 통과
+if ! echo "$BASH_CMD" | grep -qE '(^|&&|\|\|)\s*git push'; then
+    exit 0
+fi
+
+echo "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━"
+echo "  git push 감지 — 테스트를 먼저 실행합니다"
+echo "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━"
+
+PROJECT_ROOT=$(git rev-parse --show-toplevel 2>/dev/null || pwd)
+cd "$PROJECT_ROOT"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)